### ISSUE TITLE
DSQ status doesn't get cleared in ranking

### ISSUE THREAD

**Issue by theswix** [unitedworldwrestling/arena-public#39]
**Which application you are experiencing a problem with**
Arena 1.5.73

**Describe the bug**
DSQ status doesn't get cleared in ranking. Referees had erroneously set DSQ in scoreboard for on of the wrestlers, while there was VFO in reality. Clearing the match and setting VFO in Arena resulted in correct VFO in status & scoresheet, but DSQ remained in ranking and complete results.

**To Reproduce**
Cannot reproduce now since busy at the tournament, but have alredy ran into similar issue about a month ago at the Russian National Tournament. Will try to reproduce later. Probably we need just set DSQ in scoreboard, then just try to clear match result in Arena and set it to VFO and see what happens in results.

Please see the screenshots and pdfs attached.

**Expected behavior**
Correct clearing of DSQ status everywhere after changing it to VFO

**Screenshots**
<img width="1048" alt="76_FFT_status_OK" src="https://user-images.githubusercontent.com/54967199/120061031-61dd2b80-c085-11eb-8ebe-ca9d1b206616.png">
<img width="1058" alt="76_scoresheet" src="https://user-images.githubusercontent.com/54967199/120061038-673a7600-c085-11eb-8d0b-b68ed0ece0d5.png">
[76_ranking.pdf](https://github.com/unitedworldwrestling/arena-public/files/6564153/76_ranking.pdf)
[76_complete_results.pdf](https://github.com/unitedworldwrestling/arena-public/files/6564154/76_complete_results.pdf)

**Additional context**
It's kind of important issue, since Yaryguin tournamet is in UWW Calendar and wrong DSQ for an athlete is not very good idea... :( We have to print the final book tomorrow...

**Comment by csabavirag** [comment]
I am sorry for the late answer.
When you clear a DSQ match and then set the victory type to VFO, the wrestler will get its status changed from Disqualified to Forfeit. That's what you can see on the Athletes list (your 1st screenshot). Even if you click on the Athlete's name, you will see the Forfeit status.
However in the ranking, the DSQ status is determined based on A) the Athlete's status (in this case it's Forfeit, so this condition fails), B) if there was any fight closed by DSQ.
In your 76_complete_results.pdf the match # 298 in Round3 is also lost by DSQ, therefore the loser will be marked as DSQ in the final ranking.

Soon it will be more important to double check the victory type before confirming a match result. Arena will have another requirement implemented to automatically populate VFO for all the matches in a round robin group. So if you save a match lost by VFO, every other match of the loser will be automatically closed by VFO within the group.

It should not be a problem to revert those matches and change their status, it just adds an overhead to the secretariat team and must be careful to update all matches. 

So in this case, I believe Arena worked as expected and you should get the right VFO status for the Athlete after modifying the victory type on match # 298

**Comment by theswix** [comment]
Hello. I pretty much understood Arena's logic but there is one minor glitch with that DSQ & VFO, at least in this particular case: the problem is that the very first DSQ followed by immediate change of it to VFO (as soon as scoresheets arrived to the secretariat) happened in match # 259. 

By the time the match # 298 had started, the wrestler in question had VFO for almost an hour, but yet since she was supposed to take part in that match, Arena set DSQ for her disregarding VFO status. 

Basically Arena inherited DSQ from match 259 in the match 298 (both matches involved the same disqualified and right away "forfeigted" athlete.

**Comment by csabavirag** [comment]
The only way I could reproduce your problem is:
1. Create a weight with 4 participants and generate the matches
2. Open the first match (Match # 1) in Scoreboard and close it with VFO
3. Until I confirm the match in Arena there is no change on the bracket
4. Once I confirm, I get the remaining fights of the loser closed by VFO automatically (Match # 3 and 5)

![image](https://user-images.githubusercontent.com/10976654/120764374-5f148780-c518-11eb-9323-9efb97adbe6b.png)

So even if I accidentally close the match with wrong victory type in Scoreboard, until I confirm (accept the given victory type) in Arena, nothing is changed.

Also, if I change the result of Match # 1 from DSQ (even if it has been verified and confirmed by a secretariat) to VFO, that will not alter any other, already closed matches. There is no reason to overwrite an already completed match's result based on an altered result of another match of a specific wrestler. Thus the results still shows me DSQ for match # 3 and 5 unless I apply the same changes (DSQ->VFO) on those as well. Therefore even if the wrestler has its current status of Forfeit (due to the last updated match's status), his ranking status remains DSQ until there are matches the wrestler lost by DSQ.

**Comment by csabavirag** [comment]
Hi @theswix, do you have any further question?

**Comment by csabavirag** [comment]
Closing the issue. If you still have problems, please re-open.
